cip-CC-FEE-REMOVAL/cip-CC-FEE-REMOVAL.md

```diff
@@ -0,0 +1,276 @@
+<pre>
+  CIP:  CIP XXXX
+  Title: Canton Coin Fee Removal
+  Author:
+    Simon Meier
+  License: CC0-1.0
+  Status: Draft
+  Type: Tokenomics
+  Created: 2025-08-19
+  Approved:
+</pre>
+
+## Abstract
+
+This CIP proposes to remove Canton Coin fees and adjust holding fees so that
+users do not have to pay fees when transferring Canton Coin and application
+developers can build applications without special code to deal with these fees.
+This CIP further proposes to have users collect their validator rewards directly
+instead of indirectly via the validator operator party
+and to adjust the CC fees for creating or renewing `TransferPreapproval` contracts so
+that short-lived preapprovals (expiry < 90 days) can be created without paying CC fees.
+Thus making Canton Coin more attractive for both users and application
+developers.
+
+
+## Specification
+
+The changes in this CIP are initially implemented as changes to the Amulet Rules configuration where possible,
+and changes to the Splice code where necessary.
+In the future, the Splice code might be further simplified by completely removing the logic for handling CC fees.
+
+
+### Remove Canton Coin Fees
+
+Change the Amulet Rules configuration parameters as follows:
+
+| Configuration                               | Value     |
+|---------------------------------------------|-----------|
+| `transferConfig.createFee.fee`              | `0.0`     |
+| `transferConfig.transferFee.initialRate`    | `0.0`     |
+| `transferConfig.transferFee.steps.0._2`     | `0.0`     |
+| `transferConfig.transferFee.steps.1._2`     | `0.0`     |
+| `transferConfig.transferFee.steps.2._2`     | `0.0`     |
+| `transferConfig.lockHolderFee.fee`          | `0.0`     |
+
+Change the Daml code for rewards issuance in `AmuletRules_Transfer` such that featured app rewards
+are issued independently of whether CC usage fees were charged.
```

Comment:
Does this need to, or accidentally, affect the other choices with an embedded transfer? (i.e. these perform a transfer but do not have a descendant `AmuletRules_Transfer` exercise)

* `AmuletRules_CreateExternalPartySetupProposal`
* `AmuletRules_CreateTransferPreapproval`
* `AmuletRules_BuyMemberTraffic`
* `TransferPreapproval_Renew`

Reply:
The Daml PR for this changes the one place where rewards are issued, and also fixes these calls you list here. 

I'll fix the explanation here to also cover them.